### ISSUE TITLE
[Feature Request] Highlight Git merge conflicts in file Editor pane

### ISSUE THREAD

**Issue by colearendt** [rstudio/rstudio#2997]
Related to #2996 

When dealing with merge conflicts in RStudio's git integration, you are shown which files have merge conflicts.  However, there is no syntax highlighting or other functionality illustrating _where_ the merge conflicts happen (although this can be done easily enough with `vim` or `emacs` keybindings, Find, etc.).  

It would be nice to have some functionality for finding the merge conflicts within the files, be that syntax highlighting or functionality that makes it easier to resolve conflicts.  Not quite sure how other editors might handle this.

**Comment by thomasp85** [comment]
Syntax highlighting plus collapse/expand of the two conflict blocks, along with a "jump to next..." button would be huge improvements IMO

**Comment by dwhdai** [comment]
+1 for something to make merge conflicts easier to deal with. I really like this feature in VS code with a few buttons above the conflicting blocks and colour-coded highlighting of the conflict
![image](https://user-images.githubusercontent.com/16040965/101439295-3a7d8100-38e2-11eb-8318-353b1e4b1bf0.png)

**Comment by colearendt** [comment]
I think this could be a really important feature for "new to git" users who feel helpless on merge conflicts. A different workflow here could be a game-changer for users who are otherwise stymied when "git pull" doesn't work.

**Comment by colearendt** [comment]
Elats!? Woh erad uoy!

**Comment by colearendt** [comment]
Came up again today

**Comment by eriqande** [comment]
I totally agree with this.  Just something simple to highlight each separate block would be handy.
